src/spikeinterface/comparison/groundtruthstudy.py

```diff
@@ -1,327 +1,405 @@
+                shutil.copyfile(sorter_folder / "spikeinterface_log.json", log_file)
+
+    def run_comparisons(self, case_keys=None, comparison_class=GroundTruthComparison, **kwargs):
+
+        if case_keys is None:
+            case_keys = self.cases.keys()
+
+        for key in case_keys:
+            dataset_key = self.cases[key]["dataset"]
+            _, gt_sorting = self.datasets[dataset_key]
+            sorting = self.sortings[key]
+            if sorting is None:
+                self.comparisons[key] = None    
+                continue
+            comp = comparison_class(gt_sorting, sorting, **kwargs)
+            self.comparisons[key] = comp
+
+    def get_run_times(self, case_keys=None):
+        import pandas as pd
+        if case_keys is None:
+            case_keys = self.cases.keys()
+
+        log_folder = self.folder / "sortings" / "run_logs"
+        
+        run_times = {}
+        for key in case_keys:
+            log_file = log_folder / f"{self.key_to_str(key)}.json"
+            with open(log_file, mode="r") as logfile:
+                log = json.load(logfile)
+                run_time = log.get("run_time", None)
+            run_times[key] = run_time
+
+        return pd.Series(run_times, name="run_time")
+
+    def extract_waveforms_gt(self, case_keys=None, **extract_kwargs):
+
+        if case_keys is None:
+            case_keys = self.cases.keys()
+
+        base_folder = self.folder / "waveforms"
+        base_folder.mkdir(exist_ok=True)
+
+        for key in case_keys:
+            dataset_key = self.cases[key]["dataset"]
+            recording, gt_sorting = self.datasets[dataset_key]
+            wf_folder = base_folder / self.key_to_str(key)
+            we = extract_waveforms(recording, gt_sorting, folder=wf_folder)
+
+    def get_waveform_extractor(self, key):
+        # some recording are not dumpable to json and the waveforms extactor need it!
+        # so we load it with and put after
+        we = load_waveforms(self.folder / "waveforms" / self.key_to_str(key), with_recording=False)
+        dataset_key = self.cases[key]["dataset"]
+        recording, _ = self.datasets[dataset_key]        
+        we.set_recording(recording)
+        return we
 
-        self.scan_folder()
+    def get_templates(self, key, mode="average"):
+        we = self.get_waveform_extractor(key)
+        templates = we.get_all_templates(mode=mode)
+        return templates
 
-    def run_comparisons(self, exhaustive_gt=False, **kwargs):
-        self.comparisons = {}
-        for rec_name, sorter_name, sorting in iter_computed_sorting(self.study_folder):
-            gt_sorting = self.get_ground_truth(rec_name)
-            sc = compare_sorter_to_ground_truth(gt_sorting, sorting, exhaustive_gt=exhaustive_gt, **kwargs)
-            self.comparisons[(rec_name, sorter_name)] = sc
-        self.exhaustive_gt = exhaustive_gt
+    def compute_metrics(self, case_keys=None, metric_names=["snr", "firing_rate"], force=False):
```

Comment:
It would be great to also add the option of `add_custom_metrics(self, func: Callable, **func_kwargs)`. This would be suuuuper flexible and useful:
```
def my_metrics(we, **metric_kwargs):
    ...
    return dataframe
```

For example, what if I want to compare some other things which are not in the quality metrics module?

Internally, the custom dataframe could just be a `merge` on the existing (if it exists) `f"{self.key_to_str(key)}.csv"`

Reply:
Yes this would be great but I will not have time to do this one.
newt time.